libs/net/dal/Migrations/1.0.52/Up/PostUp/00-ChartTemplate.sql

```diff
@@ -0,0 +1,185 @@
+        "series" => c.Series?.Name,
+        "source" => c.OtherSource,
+        _ => c.OtherSource,
+      },
+      c => c,
+      (k, c) => new {
+        Label = k,
+        Average = c.GroupBy(c => c.TonePools.FirstOrDefault()?.Value ?? 0).Average(c => c.Key),
+        Group = c.GroupBy(c => c.TonePools.FirstOrDefault()?.Value ?? 0)
+      })
+    .OrderBy(g=> g.Label)
+    .ToArray();
+}
+{
+  "labels": [@String.Join(",", groups.Select(g => $"\"{g.Label}\""))],
+  "datasets": [
+    {
+      "label": "Negative",
+      "data": [@String.Join(",", groups.Select(g => g.Average < 0 ? Math.Round(g.Average) : 0))],
+      "backgroundColor": "red"
+    },
+    {
+      "label": "Positive",
+      "data": [@String.Join(",", groups.Select(g => g.Average > 0 ? Math.Round(g.Average) : 0))],
+      "backgroundColor": "green"
+    }
+   ]
+}' -- template
+  , '{"graph": [], "groupBy": ["product", "source", "series", "byline", "contentType"], "options": {"scales": {"x": {"max": 5, "min": -5, "ticks": {"stepSize": 1}, "stacked": true}, "y": {"ticks": {"stepSize": 1}, "stacked": true}}, "plugins": {"datalabels": {"anchor": "center", "labels": {"title": null}}}, "indexAxis": "y"}, "chartTypes": ["bar", "line", "pie", "doughnut", "polarArea", "radar"]}' -- settings
+  , '' -- created_by
+  , '' -- updated_by
+), (
+  'Earned Media' -- name
+  , 'A chart with earned media on the x-axis.' -- description
+  , false -- is_enabled
+  , true -- is_public
+  , 0 -- sort_order
+  , '@inherits RazorEngineCore.RazorEngineTemplateBase<TNO.Services.Reporting.Models.ChartTemplateModel>
+@using System
+@using System.Linq
+@using TNO.Entities
+@{
+  var group = Settings.GroupBy switch
+  {
+    "product" => Content.GroupBy(c => c.Product?.Name).OrderBy(group => group.Key),
+    "contentType" => Content.GroupBy(c => c.ContentType.ToString()).OrderBy(group => group.Key),
+    "byline" => Content.GroupBy(c => c.Byline).OrderBy(group => group.Key),
+    "series" => Content.GroupBy(c => c.SeriesId.ToString()).OrderBy(group => group.Key),
+    "sentiment" => Content.GroupBy(c => c.TonePools.FirstOrDefault()?.Value.ToString() ?? "0").OrderBy(group => group.Key),
+    "source" => Content.GroupBy(c => c.OtherSource).OrderBy(group => group.Key),
+    _ =>  Content.GroupBy(c => c.OtherSource).OrderBy(group => group.Key),
+  };
+}
+{
+  "labels": [@String.Join(",", group .Select(c => $"\"{c.Key}\""))],
+  "datasets": [
+    {
+      "label": "Totals",
+      "data":  [@String.Join(", ", group .Select(c => c.Count()))]
+    }
+   ]
+}' -- template
+  , '{"graph": [], "groupBy": ["product", "series", "source", "byline", "contentType", "sentiment"], "options": {"scales": {"x": {"ticks": {"stepSize": 1}}, "y": {"ticks": {"stepSize": 1}}}, "plugins": {"datalabels": {"anchor": "center", "labels": {"title": null}}}, "indexAxis": "y"}, "chartTypes": ["bar", "line", "pie", "doughnut", "polarArea", "radar"]}' -- settings
+  , '' -- created_by
+  , '' -- updated_by
+)
+ON CONFLICT DO NOTHING;
```

Comment:
Neat way to ignore insert errors.  Not certain this is ideal, but it works for our situation.  Currently no one has created any chart templates that are important.